source/core/src/main/com/csse3200/game/components/tasks/UseItemTask.java

```diff
@@ -0,0 +1,86 @@
+package com.csse3200.game.components.tasks;
+
+import com.csse3200.game.ai.tasks.DefaultTask;
+import com.csse3200.game.ai.tasks.PriorityTask;
+import com.csse3200.game.components.player.PlayerInventoryDisplay;
+import com.csse3200.game.entities.Entity;
+import com.csse3200.game.events.EventHandler;
+import com.csse3200.game.inventory.items.AbstractItem;
+import com.csse3200.game.inventory.items.ItemUsageContext;
+import com.csse3200.game.inventory.items.potions.AbstractPotion;
+import com.csse3200.game.inventory.items.potions.AttackPotion;
+import com.csse3200.game.inventory.items.potions.DefensePotion;
+import com.csse3200.game.inventory.items.potions.SpeedPotion;
+import com.csse3200.game.services.GameTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+public class UseItemTask extends DefaultTask implements PriorityTask {
+    private final Entity target;
+    private AbstractItem potion;
```

Comment:
Make relevant fields final. Also this should be generic for any class which has an effect over some duration - use name item rather than potion, and get the duration from the item rather than as a static constant.